File: src/main/java/quickus/core/QuickusSettings.java
package quickus.core;

import java.util.Collection;
import java.util.LinkedList;
import java.util.List;

public class QuickusSettings {
    protected final Collection<Object> stepCandidates = new LinkedList<>();
    protected final Collection<Object> featureCandidates = new LinkedList<>();

    public Collection<Object> getStepCandidates() {
        return stepCandidates;
    }

    public Collection<Object> getFeatureCandidates() {
        return featureCandidates;
    }

    public QuickusSettings withSteps(Object... steps) {
        return withSteps(List.of(steps));
    }

    public QuickusSettings withSteps(Collection<Object> steps) {
        stepCandidates.addAll(steps);
        return this;
    }

    public QuickusSettings withFeatures(Object... features) {
        return withFeatures(List.of(features));
    }

    public QuickusSettings withFeatures(Collection<Object> features) {
        featureCandidates.addAll(features);
        return this;
    }
}

File: src/main/java/quickus/core/QuickusSettings.kt
package quickus.core

import java.util.LinkedList
import quickus.core.QuickusSettings
import java.util.List

class QuickusSettings {
    private val stepCandidates = mutableListOf<Any>()
    private val featureCandidates = mutableListOf<Any>()

    fun getStepCandidates(): Collection<Any> {
        return stepCandidates
    }

    fun getFeatureCandidates(): Collection<Any> {
        return featureCandidates
    }

    fun withSteps(vararg steps: Any): QuickusSettings {
        stepCandidates.addAll(steps)
        return this
    }

    fun withFeatures(vararg features: Any): QuickusSettings {
        featureCandidates.addAll(features)
        return this
    }
}
